libvirt/domain.go

```diff
@@ -78,31 +78,22 @@ func domainWaitForLeases(domain *libvirt.Domain, waitForLeases []*libvirtxml.Dom
 	return err
 }
 
-func domainIfaceHasAddress(domain libvirt.Domain, iface libvirtxml.DomainInterface,
-	domainDef libvirtxml.Domain, virConn *libvirt.Connect, rd *schema.ResourceData) (found bool, ignore bool, err error) {
+func domainIfaceHasAddress(domain libvirt.Domain, domainDef libvirtxml.Domain, virConn *libvirt.Connect, rd *schema.ResourceData) (found bool, ignore bool, err error) {
 
-	mac := strings.ToUpper(iface.MAC.Address)
```

Comment:
why this removal:

with the new behaviour we are assigning an empty MAC to network interfaces.

The `iface` here is the temporary interface.
https://github.com/dmacvicar/terraform-provider-libvirt/pull/422/files#diff-2f03be069459ebc91fcf0607d4d686d8R1316

With `domainGetIfacesInfo(domain, domainDef, virConn, rd)`  we get the `real` interfaces so we have the MAC adresses there.